File: applications/EmployeeNetCoreApp/EmployeeNetCoreApp/Model/Department.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EmployeeNetCoreApp.Model
{
	[Table("Departments")]
	public class Department
	{
        [Key]
        public int DepartmentId { get; set; }
        [Required]
        public string? DepartmentName { get; set; }
        [ForeignKey("Department")]
        public ICollection<Employee>? Employees { get; set; }
    }
}

File: applications/EmployeeQuarkusApp/src/main/java/com/redhat/developers/model/Department.java
package com.redhat.developers.model;

import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.SequenceGenerator;
import javax.validation.constraints.NotEmpty;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Department {

    @Id
    @SequenceGenerator(name = "departmentSeq", sequenceName = "department_id_seq", allocationSize = 1, initialValue = 1)
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "departmentSeq")
    private Long departmentId;
    @NotEmpty
    @Column(nullable = false, unique = true)
    private String departmentName;

    @OneToMany(mappedBy = "department")
    private List<Employee> employees;
}
